test/server/cards/02_SHD/units/SlaversFreighter.spec.ts

```diff
@@ -0,0 +1,47 @@
+describe('Slaver\'s Freighter', function() {
+    integration(function(contextRef) {
+        describe('Slaver\'s Freighter\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['slavers-freighter'],
+                        groundArena: [{ card: 'cell-block-guard', exhausted: true }, { card: 'count-dooku#darth-tyranus', exhausted: true }],
+                    },
+                    player2: {
+                        groundArena: [{ card: 'wampa', upgrades: ['fallen-lightsaber', 'electrostaff'], exhausted: true }],
+                        spaceArena: [{ card: 'green-squadron-awing', upgrades: ['resilient'] }]
+                    }
+                });
+            });
+
+            it('should allow a friendly unit with power less than or equal to the number of upgrades on enemy units to be readied.', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.slaversFreighter);
+
+                expect(context.player1).toHavePassAbilityButton();
+                expect(context.player1).toBeAbleToSelectExactly([context.cellBlockGuard, context.greenSquadronAwing]);
+
+                context.player1.clickCard(context.cellBlockGuard);
+
+                expect(context.cellBlockGuard.exhausted).toBe(false);
+
+                expect(context.player2).toBeActivePlayer();
+            });
+
+            it('should be optional', function () {
```

Comment:
We have a practical issue with our tests where the Jasmine tests are super slow and we have finite Github Actions time in the free tier, so we have to be pretty sparing in our tests and try to consolidate down as much as we can (even though it's not great test practice obviously).

In this case we typically don't add a whole test for the pass ability button, our practice is to assume that if it's present it's working (and we do have tests that check that).